public/src/pages/TeacherPage.tsx

```diff
@@ -348,32 +350,25 @@ export class TeacherPage extends ViewPage {
         }
     }
 
-    private async getCourses(): Promise<Course[]> {
+    private async getCourses(statuses: Enrollment.UserStatus[]): Promise<Course[]> {
         const curUsr = this.userMan.getCurrentUser();
         if (curUsr) {
-            return this.courseMan.getCoursesForUser(curUsr, []);
+            const courses = await this.courseMan.getCoursesForUser(curUsr, statuses);
+            const enrols = await this.courseMan.getEnrollmentsForUser(curUsr.getId());
+            return getActiveCourses(courses, enrols, curUsr.getId());
         }
         return [];
     }
 
     private async getCourseURL(courseID: number): Promise<string> {
-        const repoMap = await this.courseMan.getRepositories(
-            courseID,
-            [Repository.Type.COURSEINFO],
-            );
-        const fullRepoName = repoMap.get(Repository.Type.COURSEINFO);
-        return fullRepoName ? fullRepoName.split("course-info")[0] : "";
+        const repoMap = this.repositories.get(courseID);
+        return repoMap?.get(Repository.Type.COURSEINFO) ?? "";
     }
 
     private async courseFunc(courseParam: string, fn: (course: Course) => View): View {
         const courseId = parseInt(courseParam, 10);
-        const course = await this.courseMan.getCourse(courseId);
```

Comment:
`courseId` on line 369 is no longer needed

Reply:
And similarly for `courseParam`

Reply:
I also don’t think you need the function parameter name `course` in `(course: Course)`

Reply:
Fixed those too.